In [16]:
# 4011 : 생년월일 출력

ssn = input()
n = []
for i in ssn:
    if i != "-":
        n.append(i)

if n[6] == "1":
    print("19{}{}/{}{}/{}{} M".format(n[0],n[1],n[2],n[3],n[4],n[5]))
else:
    print("19{}{}/{}{}/{}{} F".format(n[0],n[1],n[2],n[3],n[4],n[5]))
    

781201-1273845
1
1978/12/01 M


In [20]:
# 4013 : 진법 변환

n = int(input())

print("2 {:0b}".format(n))
print("8 {:0o}".format(n))
print("16 {:0X}".format(n))

15
2 1111
8 17
16 F


In [28]:
# 4016 : 세 수의 최대공약수 구하기

arr = list(map(int, input().split()))

def twonum(a, b):
    while b != 0:
        a, b = b, a%b
    return a

a_arr = arr[0]
for i in range(len(arr)):
    a_arr = twonum(a_arr, arr[i])
    
print(a_arr)

4 8 10
2


In [36]:
# 4021 : 홀수의 합 구하기

num = list(map(int, input().split()))
total = 0
flag = False
for i in num:
    if i % 2 == 1:
        total += i
        flag = True

if flag == False:
    print(-1)
else:
    print(total)

12 12 12 12 12 12 12
-1


In [62]:
# 4026 : 중앙 값

n = list(sorted(map(int, input().split())))

if len(n) % 2 == 0:
    x = (n[(len(n)//2)-1] + n[(len(n)//2)]) / 2
    print(x)
else:
    x = n[(len(n)//2)]
    print(x)

40 40 60 20 90
40


In [7]:
# 4031 : 가장 큰 수

num = map(int, input().split())
x = []
y = []
for i in num:
    if i % 2 == 0:
        x.append(i)
    if i % 2 == 1:
        y.append(i)
if len(x) == 0:
    print(max(y))
elif len(y) == 0:
    print(max(x))
else:
    print(max(x) + max(y))

1 1 1 1 1 1 1
1


In [26]:
# 4036 : 합과 차

n = int(input())
m = int(input())
x = (n+m) // 2
y = (n-m) // 2


for i in range(x+1):
    if (x*2) == (n + m):
        for j in range(y+1):
            if (y*2) == (n - m):
                break
            else:
                y += 1
    else:
        x += 1

print(x)
print(y)

1000000
999998
999999
1


In [59]:
# 4041 : 숫자 다루기
n = input()
k = []
j = []
for i in range(len(n), 0, -1):
    k.append(n[i-1])
    j.append(int(n[i-1]))
f = "".join(k)
print(int(f))
print(sum(j))

320
23
5


In [23]:
# 4051 : 시간외 근무 수당
se = [list(map(float,input().split())) for i in range(5)]
x = []
for i in range(len(se)):
    x.append(se[i][1] - se[i][0])
if sum(x) <= 5:
    print((sum(x)-1)*5000 + (sum(x)-1)*5000*0.05)
elif sum(x) >= 15:
    print((sum(x)-1)*5000 - (sum(x)-1)*5000*0.05)
else:
    print((sum(x)-1)*5000)
    
# 중단

16.0 16.5
17.0 20.5
18.5 23.0
22.5 23.5
19.5 22.5
57500.0


In [35]:
# 4056 : 연말 정산
import math

n = int(input())

if n <= 500:
    x = n * 0.7
    print(math.floor(x))
elif n > 500 and n <= 1500:
    x = 350 + (n - 500) * 0.4
    print(math.floor(x))
elif n > 1500 and n <= 4500:
    x = 750 + (n - 1500) * 0.15
    print(math.floor(x))
elif n > 4500 and n <= 10000:
    x = 1200 + (n - 4500) * 0.05
    print(math.floor(x))
else:
    x = 1475 + (n - 10000) * 0.02
    print(math.floor(x))

3130
994


In [41]:
# 4501 : 백설공주와 난장이
n = []

for i in range(7):
    x = int(input())
    n.append(x)
n = sorted(n, reverse=True)
for i in range(2):
    print(n[i])

79
57
88
72
95
88
64
95
88


In [58]:
# 4506 : 최대공약수와 최소공배수
n, m = map(int, input().split())

def twonum(a, b):
    while b != 0:
        a, b = b, a%b
    return a
    
x = twonum(n,m)
print(x)

def mulnum(a, b, x):
    y = a * b // x
    return y
y = mulnum(n,m,x)

print(y)

24 18
6
72


In [63]:
# 4531 : 대표 값

n = []
for i in range(5):
    k = int(input())
    n.append(k)
n = sorted(n)
y = sum(n)//len(n)
print(y)

if len(n) % 2 == 0:
    x = (n[(len(n)//2)-1] + n[(len(n)//2)]) / 2
    print(x)
else:
    x = n[(len(n)//2)]
    print(x)

40
40
60
20
90
50
40


In [20]:
# 4536 : 대표값
from collections import Counter

n = []
for i in range(10):
    k = int(input())
    n.append(k)
n = sorted(n)
y = sum(n)//len(n)
print(y)
s = Counter(n).most_common()
print(s[0][0])

10
40
30
60
30
20
60
30
40
50
37
30


[문제1] 시간외 근무 수당 (16점, 제한시간 1초)

A대학에서는 시간외 근무를 수기로 작성하고 있다. 행정실의 K씨는 시간외 근무 수당을 전산으로 처리하고자 한다. 이 대학의 시간외 근무 수당 규정은 아래와 같다.

1. 시간외 근무는 최초 1시간은 제외하며, 최대 4시간까지 인정한다.

2. 시간외 근무에 따른 금액은 30분에 5,000원으로 한다.

3. 5일간의 시간외 근무 합계가 15시간 이상이면 받는 금액의 5%를 감하여 지급한다.

4. 5일간의 시간외 근무 합계가 5시간 이하이면 받는 금액의 5%를 더하여 지급한다.

시간외 근무 수당을 계산하는 프로그램을 작성하시오.

입력
1. 5일간의 시간외 근무 시작 시각 s와 종료 시각 e가 주어진다. (단, 16.0 ≦ s ≦ e ≦ 24.0)

2. 30분 단위로 입력이 가능하며, 30분은 0.5로 입력된다. (16시 30분은 16.5로 표기한다.)

출력
첫 줄에 최종 시간외 근무 수당을 출력한다(단, 단위는 생략한다).

입력 예시   
16.0 16.5
17.0 20.5
18.5 23.0
22.5 23.5
19.5 22.5

출력 예시
80000

In [37]:
# 4051 : 시간외 근무 수당

n = [[] for i in range(5)]

for i in range(5):
    s, e = map(float, input().split())
    n[i].append(s)
    n[i].append(e)
    
t_sum = 0
money = 0
for i in n:
    if (i[1] - i[0] > 1) and (i[1] - i[0] <= 5) :
        t_sum += i[1] - i[0] - 1
    elif (i[1] - i[0] > 5) :
        t_sum += 4

if t_sum >= 15 :
    t_sum = t_sum / 0.5
    money = t_sum * 5000
    money -= (money * 0.05)
elif t_sum <= 5 :
    t_sum = t_sum / 0.5
    money = t_sum * 5000
    money += (money * 0.05)
else :
    t_sum = t_sum / 0.5
    money += t_sum * 5000
    
print(int(money))

16.0 21.0
16.5 21.5
17.5 21.5
19.5 24.0
22.5 24.0
142500
